File: src/main/java/com/developersam/web/control/filter/DirectToIndexFilter.java
package com.developersam.web.control.filter;

import javax.servlet.Filter;
import javax.servlet.FilterChain;
import javax.servlet.FilterConfig;
import javax.servlet.ServletException;
import javax.servlet.ServletRequest;
import javax.servlet.ServletResponse;
import javax.servlet.annotation.WebFilter;
import javax.servlet.http.HttpServletRequest;
import java.io.IOException;

@WebFilter(filterName = "DirectToIndexFilter")
public class DirectToIndexFilter implements Filter {
    
    @Override
    public void init(FilterConfig filterConfig) { }
    
    @Override
    public void doFilter(ServletRequest req, ServletResponse resp,
                         FilterChain chain)
            throws ServletException, IOException {
        req.getRequestDispatcher("index.html").forward(req, resp);
    }
    
    @Override
    public void destroy() {}
    
}

File: src/main/kotlin/com/developersam/controller/filter/DirectToIndexFilter.kt
package com.developersam.controller.filter

import javax.servlet.Filter
import javax.servlet.FilterChain
import javax.servlet.FilterConfig
import javax.servlet.ServletRequest
import javax.servlet.ServletResponse
import javax.servlet.annotation.WebFilter

/**
 * A filter that works with Angular to redirect some sources to index.html
 * so that Angular's routing system can be properly used.
 */
@WebFilter(filterName = "DirectToIndexFilter")
class DirectToIndexFilter : Filter {

    override fun init(filterConfig: FilterConfig) {}

    override fun doFilter(req: ServletRequest, resp: ServletResponse,
                          chain: FilterChain) {
        req.getRequestDispatcher("index.html").forward(req, resp)
    }

    override fun destroy() {}

}
